LFGBulletinBoard/LibGPIMinimapButton.lua

```diff
@@ -11,36 +17,43 @@ local function BottomZoom(button)
 	button.Lib_GPI_MinimapButton.icon:SetTexCoord( deltaX, 1 - deltaX,  deltaY, 1 - deltaY)
 end
 
+-- Calculates the the angle and distance of the minimap button based on cursor position about the minimap center.
+---@param button MinimapButton
 local function onUpdate(button)
 	local mx, my = Minimap:GetCenter()
 	local px, py = GetCursorPosition()
-	local w=((Minimap:GetWidth() / 2) + 5)
 	local scale = Minimap:GetEffectiveScale()
-	px, py = px / scale, py / scale
-	local dx,dy= px - mx,py - my
-	local dist=math.sqrt(dx*dx+dy*dy)/w
+	px, py = (px / scale), (py / scale);
+	local dx, dy = (px - mx), (py - my);
+
+	-- calculate the angle of the button based on the cursor position offset from the minimap center
+	button.Lib_GPI_MinimapButton.db.position = math.deg(math.atan2(dy, dx)) % 360
+
+	-- calculate a distance ratio for the position of the button based on radius.
+	-- ie `2` means the button is located 2x the size of the radius away from the center
 	if button.Lib_GPI_MinimapButton.db.lockDistance then
-		dist=1
+		button.Lib_GPI_MinimapButton.db.distance = 1 -- locked the perimeter of minimap
 	else
-		if dist<1 then dist=1 elseif dist>2 then dist=2 end
+		local radius = (Minimap:GetWidth() / 2) + 5;
+		local dist = math.sqrt(dx*dx+dy*dy) / radius
+		button.Lib_GPI_MinimapButton.db.distance = Clamp(dist, 1, 2) -- clamp the distance ratio between 1 and 2
```

Comment:
moved this math into the conditional to only be used when necessary (`onUpdate` functions run every frame)